### ISSUE TITLE
Add license

### ISSUE THREAD

**Issue by zorggn** [love2d-community/love-api#77]
In the löve discord, someone brought this issue up since they can't incorporate this into their project since they cannot provide a license for it.

**Comment by ITR13** [comment]
Any updates on this?

**Comment by hahawoo** [comment]
The information is from the wiki, and I assume the wiki documentation is under the GNU Free Documentation License because it says "Content is available under GNU Free Documentation License 1.3 unless otherwise noted." at the bottom of the wiki pages. You might want to ask one of the devs to clarify this. As far as I'm concerned, everything I've done on this is public domain, but maybe this legally needs to be under the GNU Free Documentation License too.

**Comment by rm-code** [comment]
I asked about this wayyyy back (probably a few years by now) on the Discord channel, but didn't get a clear answer...

I'd say the save thing to do is to add the GNU License.

(Then again there are also references to the FreeBSD license, but they are probably outdated).

**Comment by pablomayobre** [comment]
FreeBSD is outdated yeah, nothing remains of those docs.

GNU Free Documentation License should be used, considering the amount of data taken straight from the wiki.

Also all projects that use it should probably expose the JSON table in some way in order to avoid the GNU Free Documentation license affecting the entire project (cause it's invasive as f*ck)

**Comment by rm-code** [comment]
What do you mean by "expose the JSON table"?

**Comment by pablomayobre** [comment]
Say for love-atom you created the derivative `.json` table, that's under GNU Free Documentation License too, and so is the Github Pages HTML version, and any other plugin that were to use this as a source because the license is as invasive (or worse) than GPL

**Comment by MikuAuahDark** [comment]
The JSON expose table is pretty much solved (https://love2d-community.github.io/love-api/love-api.json). However the question of the license still remains.

**Comment by mark-wiemer** [comment]
There's the question of the license of the content, but I'm curious about the license of this repo itself. What is the license of the authored code here, the code that scrapes and publishes the API?

**Comment by MikuAuahDark** [comment]
Yeah the license is complicated to solve.

LOVE Wiki is licensed under GNU FDL 1.3 which means the whole Lua file in here is subject to that license and it has its own gotchas. Considering that someone also mentioned that the FDL is a viral license, this means the JSON output is also gonna be in FDL too. This also means, for an example, definitions file generated by LuaLS using the JSON file also falls under FDL. The legally questionable part is if the processing program should also fall into FDL.

Ideally we'd like to use Creative Commons, however asking contributors to relicense is time consuming, if it's even possible (e.g. controbutions made by deceased person). Core LOVE developers like Sasha and bartbes likely open to relicense it. I also personaly don't mind relicensing my contributions to another license. Again, the issue is contributions by other people which may needs their permission (with above issues), rewritten, or removed.

In short, the situation is complicated.

**Comment by mark-wiemer** [comment]
I understand that licensing the output is a bit difficult. However, do you have any thoughts on the license of the web scraper itself? If somebody wanted to take that part of the repo and republish it, what license would they be able to use?

**Comment by MikuAuahDark** [comment]
I believe the web scraper is only used as building blocks. Nowadays the changes are hand-copied or hand-written from the wiki.

You'd probably have to ask the original author of the original scraper script for that, unfortunately.

**Comment by thomashope** [comment]
@MikuAuahDark the wiki was contributed by @hahawoo who indicated above that they consider their contribution to be in the public domain.

AFAIK the wiki scraper was discussed in issue #84 and committed directly to master in commit 7c8e562

Also, while I am not a lawyer, my understanding is it's perfectly valid to have different parts of a project under different licences.

i.e. the whole repo could be under license A, with the exception of those files containing content from the wiki under licence B. This could be made clear in the repo readme + any exceptional files could have their licence clarified at the top of the file.

**Comment by ITR13** [comment]
> Also, while I am not a lawyer, my understanding is it's perfectly valid to have different parts of a project under different licences.
> 
> i.e. the whole repo could be under license A, with the exception of those files containing content from the wiki under licence B. This could be made clear in the repo readme + any exceptional files could have their licence clarified at the top of the file.

It depends on the license, some licenses are self-perpetuating and would require the whole repo to be licensed with a similar license, see [Copyleft#Reciprocity](https://en.wikipedia.org/wiki/Copyleft#Reciprocity) and [Copyleft#Strong_and_weak_copyleft](https://en.wikipedia.org/wiki/Copyleft#Strong_and_weak_copyleft). There's nothing preventing somebody from having their original work be additionally licensed as a public domain license (I.E. public domain), but the work as a whole and contributions to already existing material would need to be FDL licensed too.

But also not a lawyer, so possible I got some parts wrong :)
